webapp/src/services/api/studydata.ts

```diff
@@ -61,9 +61,12 @@ export const deleteLink = async (
 };
 
 export const getAllLinks = async (
-  uuid: string
+  uuid: string,
+  withUi?: boolean
 ): Promise<Array<LinkCreationInfo>> => {
```

Comment:
Use conditional types return (cf `getUsers` in "src/services/api/users.ts").
You need to have `LinkCreationInfo` and `LinkCreationInfoWithUI` that extends the previous one.